src/ophyd_async/epics/pmac/_pmacTrajectory.py

```diff
@@ -0,0 +1,128 @@
+            self.profile[axis.lower()] = []
+            if axis != "DURATION":
+                self.profile[axis.lower() + "_velocity"] = []
+
+        # Calc Velocity
+
+        for axis in scanAxes:
+            for i in range(scanSize - 1):
+                if axis != "DURATION":
+                    self.profile[axis.lower() + "_velocity"].append(
+                        (
+                            scanSpecStack[0].midpoints[axis][i + 1]
+                            - scanSpecStack[0].midpoints[axis][i]
+                        )
+                        / (scanSpecStack[0].midpoints["DURATION"][i])
+                    )
+                    self.profile[axis.lower()].append(
+                        scanSpecStack[0].midpoints[axis][i]
+                    )
+                else:
+                    self.profile[axis.lower()].append(
+                        int(scanSpecStack[0].midpoints[axis][i] / TICK_S)
+                    )
+            if axis != "DURATION":
+                self.profile[axis.lower()].append(
+                    scanSpecStack[0].midpoints[axis][scanSize - 1]
+                )
+                self.profile[axis.lower() + "_velocity"].append(0)
+            else:
+                self.profile[axis.lower()].append(
+                    int(scanSpecStack[0].midpoints[axis][scanSize - 1] / TICK_S)
+                )
+
+        # Calculate Starting Position to allow ramp up to velocity
+        self.initial_pos = {}
+        run_up_time = 0
+        for axis in scanAxes:
+            if axis != "DURATION":
+                run_up_disp, run_up_time = await self._ramp_up_velocity_pos(
+                    0,
+                    self.motors[axis.lower()],
+                    self.profile[axis.lower() + "_velocity"][0],
+                )
+                self.initial_pos[axis] = self.profile[axis.lower()][0] - run_up_disp
+        self.profile["duration"][0] += run_up_time / TICK_S
+
+        # Send trajectory to brick
+        for axis in scanAxes:
+            if axis != "DURATION":
+                self.profile_cs_name.set(self.cs)
+                self.points_to_build.set(scanSize)
+                getattr(self, "use_" + axis.lower()).set(True)
+                getattr(self, axis.lower()).set(self.profile[axis.lower()])
+                getattr(self, axis.lower() + "_vel").set(
+                    self.profile[axis.lower() + "_velocity"]
+                )
+            else:
+                self.timeArray.set(self.profile["duration"])
+
+        # MOVE TO START
+        for axis in scanAxes:
+            if axis != "DURATION":
+                await self.motors[axis.lower()].set(self.initial_pos[axis])
+
+        # Set No Of Points
+
+        self.build_profile.set(True)
+        self._fly_start = time.monotonic
+
+    @AsyncStatus.wrap
+    async def kickoff(self):
+        await self.execute_profile.set(True)
+
+    @WatchableAsyncStatus.wrap
+    async def complete(self):
+        yield WatcherUpdate(
```

Comment:
This should be:
```python
async for percent in observe_value(self.percent_complete):
    yield WatcherUpdate(..., current=percent, ...)
```
Eventually we will need to use trajectory points rather than percent, but that can come later